File: 프로그래머스/1/12926. 시저 암호/시저 암호.js
const solution = (s, n) => {
  const upperalpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZABCDEFGHIJKLMNOPQRSTUVWXYZ";
  const loweralpha = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz";

  let answer = "";
  let iii;

  for (let i of s) {
    let iii = 0;

    if (i === i.toLowerCase() && i !== " ") {
      iii = loweralpha.indexOf(i);

      answer += loweralpha.charAt(iii + n);
    } else if (i === i.toUpperCase() && i !== " ") {
      iii = upperalpha.indexOf(i);

      answer += upperalpha.charAt(iii + n);
    } else {
      answer += " ";
    }
  }
  return answer;
};

File: 프로그래머스/1/12926. 시저 암호/시저 암호.java
class Solution {
    public String solution(String s, int n) {
        String answer = "";
        String lowerAlpha = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz" ; 
        String upperAlpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZABCDEFGHIJKLMNOPQRSTUVWXYZ" ; 
        
        
        int idx ; 
        
        for(int i =0 ; i< s.length(); i++){
            char curr =  s.charAt(i);
            //소문자 일 때 
            if(curr==Character.toLowerCase(curr) && curr!=' ' ){ 
                 idx = lowerAlpha.indexOf(curr) ; 
                answer+= lowerAlpha.charAt(idx+n) ; 
                
                    
            }
            else if(curr==Character.toUpperCase(curr) && curr!=' '){
                 idx = upperAlpha.indexOf(curr) ; 
                answer+= upperAlpha.charAt(idx+n) ; 
                
            }
            else{
                answer+=" " ; 
            }
        }
        return answer;
    }
}